File: mergeSortedArray.cpp
#include "leetcode.h"
class Solution {
public:
    void merge(int A[], int m, int B[], int n) {
        int C[m+n];
        int i = 0;
        int j = 0;
        int k = 0;
        while (i < m && j < n) {
            if (A[i] < B[j]) {
                C[k++] = A[i++];
            }
            else {
                C[k++] = B[j++];
            }
        }
        while (i < m) {
            C[k++] = A[i++];
        }
        while (j < n) {
            C[k++] = B[j++];
        }
        k = 0;
        i = 0;
        while (k < (m+n)) {
            A[i++] = C[k++];
        }
    }
};

File: Algorithms/mergeSortedArray.py
class Solution:
    # @param A  a list of integers
    # @param m  an integer, length of A
    # @param B  a list of integers
    # @param n  an integer, length of B
    # @return nothing(void)
    def merge(self, A, m, B, n):
        i, j = 0, 0
        C = []
        while i < m and j < n:
            if A[i] < B[j]:
                C.append(A[i])
                i += 1
            else:
                C.append(B[j])
                j += 1
        if i < m:
            C.extend(A[i:])
        if j < n:
            C.extend(B[j:])
        for i in range(m+n):
            A[i] = C[i]
